File: examples/swift-client/Sources/AppwriteModels/UserList.swift

/// Users List
public class UserList {

    /// Total number of items available on the server.
    public let sum: Int

    /// List of users.
    public let users: [User]

    init(
        sum: Int,
        users: [User]
    ) {
        self.sum = sum
        self.users = users
    }

    public static func from(map: [String: Any]) -> UserList {
        return UserList(
            sum: map["sum"] as! Int,
            users: (map["users"] as! [[String: Any]]).map { User.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "sum": sum as Any,
            "users": users.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                    
}

File: examples/flutter/lib/src/models/user_list.dart
part of appwrite.models;

/// Users List
class UserList {
    /// Total number of items available on the server.
    final int sum;
    /// List of users.
    final List<User> users;

    UserList({
        required this.sum,
        required this.users,
    });

    factory UserList.fromMap(Map<String, dynamic> map) {
        return UserList(
            sum: map['sum'],
            users: List<User>.from(map['users'].map((p) => User.fromMap(p))),
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "sum": sum,
            "users": users.map((p) => p.toMap()),
        };
    }
}
